data/Record.js

```diff
@@ -20,29 +21,44 @@ export class Record {
     id;
     /** @member {(string|number)} */
     parentId;
-    /** @member {Store} */
-    store;
+    /** @member {Record} */
+    originalRecord;
```

Comment:
might default original to point to `this`, given its name, and then use a === check for isDirty().